azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -1209,23 +1133,24 @@ def test_cancels_watchdog_on_pending_connect(self, mocker, stage, mock_timer, ca
         assert mock_timer.return_value.cancel.call_count == 1
 
     @pytest.mark.it(
-        "Cancels the connection watchdog if the pending operation is a ReauthorizeConnectionOperation"
+        "Does not cancel the connection watchdog if the pending operation is a ReauthorizeConnectionOperation"
     )
-    def test_cancels_watchdog_on_pending_reauthorize(self, mocker, stage, mock_timer, cause):
+    def test_does_not_cancel_watchdog_on_pending_reauthorize(
```

Comment:
>test_does_not_cancel_watchdog_on_pending_reauthorize [](start = 8, length = 52)

i understand what you want to test , but if you just look at the code of `_on_mqtt_disconnected` , regardless of what type of op it is we are always calling  `_cancel_connection_watchdog` , so when the test becomes does not cancel , it is not matching if we just read the code.

Reply:
For disconnect and reauthorize, there is no connection watchdog, so there's nothing to cancel.  The call to `_cancle_connection_watchdog` just doesn't do anything in this case.  I changed the test name to be a little more clear and, unfortunately, also a little big more awkward.

---
In reply to: [437786906](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r437786906) [](ancestors = 437786906)